pkg/apis/performanceprofile/v2/performanceprofile_validation.go

```diff
@@ -218,14 +228,14 @@ func (r *PerformanceProfile) validateHugePages() field.ErrorList {
 	// validate that default hugepages size has correct value, currently we support only 2M and 1G(x86_64 architecture)
 	if r.Spec.HugePages.DefaultHugePagesSize != nil {
 		defaultSize := *r.Spec.HugePages.DefaultHugePagesSize
-		if defaultSize != hugepagesSize1G && defaultSize != hugepagesSize2M {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("spec.hugepages.defaultHugepagesSize"), r.Spec.HugePages.DefaultHugePagesSize, fmt.Sprintf("hugepages default size should be equal to %q or %q", hugepagesSize1G, hugepagesSize2M)))
+		if !slices.Contains(validHugepagesSizes, defaultSize) {
```

Comment:
I think it would be great if we could validate that the page size you have selected is valid for your hardware architecture, but I don't think it would be very easy

Hypothetically NTO can detect the architecture using its existing ghw library, but that would only work on the node where NTO is installed/running

In a distributed model NTO could be running on an x86 hub, but managing aarch64 spokes and I'm not sure if there's a way for us to validate that

Reply:
we do have node arch exposed by core k8s as label:
```
$ oc get nodes MYNODE -o json | jq .metadata.labels
{
  "beta.kubernetes.io/arch": "amd64",
  "beta.kubernetes.io/os": "linux",
  "kubernetes.io/arch": "amd64",
  "kubernetes.io/hostname": "MYNODE",
  "kubernetes.io/os": "linux",
  "node-role.kubernetes.io/worker": "",
  "node.openshift.io/os_id": "rhcos"
}
```
and I think I can build on that. Actually I'd go a step further and add the requirement that all the node in a machine config pool must have the same arch, which seems a pretty fair requirement to me.

Reply:
I would even add the requirement that CPU capacity is the same. Because otherwise the CPU IDs will not work for some nodes.

Reply:
@MarSik - can you please elaborate on what you mean about the CPU IDs? Is it really OK for us to force all the nodes in a machine config pool to have the same CPU capacity, even if they are the same CPU architecture? Is this something we should clear with PM before enforcing in code?